src/main.ts

```diff
@@ -53,21 +56,24 @@ async function run(): Promise<void> {
 
     output = ''
     core.info('Run Terraform Plan')
-    await exec.exec(terraformPath, ['plan', '-refresh=false', '-no-color'], options)
+    await exec.exec(terraformPath, ['plan', '-refresh=false', '-no-color', '-out=plan'], options)
 
-    if (comment && !(github.context.eventName === 'push' || github.context.payload.pull_request?.merged)) {
+    if (comment) {
       core.info('Add Plan Output as a Comment to PR')
-      await addComment(issue_number, output, githubToken, github.context, false)
+      if (github.context.eventName === 'push') {
+        premessage = 'Output from Terraform plan before Apply\n'
+      }
+      await addComment(issue_number, premessage, output, githubToken, github.context, false)
     }
 
     output = ''
-    if (github.context.eventName === 'push' || github.context.payload.pull_request?.merged) {
+    if (github.context.eventName === 'push' || github.context.eventName === 'workflow_dispatch' || github.context.eventName === 'repository_dispatch') {
```

Comment:
If push, don't you need to check the branch name is main or master.  Otherwise could trigger on any push to a branch.  In workflow/repo dispatch mode, is it worth having a variable set which controls whether to apply or not.

Reply:
I am happy with that, namely apply on any push as it is sometimes helpful to be able to run the whole thing from a feature branch.

However, maybe you are right and it should have a flag that needs to be set so that it's actually explicit